### PR TITLE
refactor(option): DRY up get/set option value

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: src/nvim/api/options.c [hunk 1]
```diff
@@ -486,66 +457,29 @@ void set_option_to(uint64_t channel_id, void *to, int type, String name, Object
...
     }
-
-    api_set_error(err, kErrorTypeException, "%s", errmsg);
+    return 0;
   }
 }
 
-int get_option_value_for(char *key, long *numval, char **stringval, int opt_flags, int opt_type,
-                         void *from, Error *err)
+static int get_or_set_option_value_for(char *key, long *numval, char **stringval, int opt_flags, int opt_type,
```
[the remark this answers is not in the document]
Comment (lewis6991) [reply src/nvim/api/options.c:481]: I agree. I was tempted to just do `option_value_for` but decided it was a little vague.

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/api/options.c:481]: Yeah, needs a verb ;) Maybe `:h dev-api` can give some ideas?

[the remark this answers is not in the document]
Comment (lewis6991) [reply src/nvim/api/options.c:481]: `return_option_value{_for}` to me seems like an alternative name for `get_option_value{_for}` and doesn't indicate the function can set options.

[the remark this answers is not in the document]
Comment (lewis6991) [reply src/nvim/api/options.c:481]:
```
Use existing common {action} names if possible:
    add     Append to, or insert into, a collection
    create  Create a new thing
    del     Delete a thing (or group of things)
    exec    Execute code
    get     Get a thing (or group of things by query)
    list    Get all things
    set     Set a thing (or group of things)
```

Hm.... I'm not sure there is an ideal solution here.

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/api/options.c:481]: Hmm... "Do two things" is not a great function ;) (That is something to keep in mind: right now, getting and setting share this code, but will this necessarily be the case? Does the refactor make it _harder_ to implement or fix something for `get` _or_ `set` in the future?)

Can you summarize in one sentence what this function does?

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/api/options.c]: `access_option_value_for`?

(But maybe I'm overthinking this. Let's wait until @justinmk wakes up and see what he thinks :))

[the remark this answers is not in the document]
Comment (lewis6991) [reply src/nvim/api/options.c]: > `access_option_value_for`?

Done

> EDIT: But one thing I can already guess at: you should put these explanations (especially about setting up and restoring context) into the commit message ;)

Done

**Review by justinmk — APPROVED** [review]
Thank you! 

Side note: wondering if we should do validation in Lua (or an elaborate C macro), to make things much less verbose.
